source/material_model/rheology/friction_options.cc

```diff
@@ -0,0 +1,495 @@
+                           "\n\n"
+                           "\\item ``default'': No rate or state dependence of the friction angle is applied. ");
+
+        // Plasticity parameters
+        /*should I do this or just read in the internal anlges of friction directly? */
+        /*drucker_prager_parameters = drucker_prager_plasticity.parse_parameters(this->n_compositional_fields()+1,
+                                                                               prm);*/
+
+        // Dynamic friction paramters
+        prm.declare_entry ("Dynamic characteristic strain rate", "1e-12",
+                           Patterns::Double (0),
+                           "The characteristic strain rate value, where the angle of friction takes the middle "
+                           "between the dynamic and the static angle of friction. When the effective strain rate "
+                           "in a cell is very high the dynamic angle of friction is taken, when it is very low "
+                           "the static angle of internal friction is chosen. Around the dynamic characteristic "
+                           "strain rate, there is a smooth gradient from the static to the dynamic friction "
+                           "angle. "
+                           "Units: $1/s$.");
+
+        prm.declare_entry ("Dynamic angles of internal friction", "9999",
+                           Patterns::List(Patterns::Double(0)),
+                           "List of dynamic angles of internal friction, $\\phi$, for background material and compositional "
+                           "fields, for a total of N+1 values, where N is the number of compositional fields. "
+                           "For a value of zero, in 2D the von Mises criterion is retrieved. "
+                           "Dynamic angles of friction are used for calculation when the effective strain rate in a cell "
+                           "is well above the characteristic strain rate. If not specified, the internal angles of "
+                           "friction are taken. "
+                           "Units: degrees.");
+
+        prm.declare_entry ("Dynamic friction smoothness exponent", "1",
+                           Patterns::Double (0),
+                           "An exponential factor in the equation for the calculation of the friction angle "
+                           "when a static and a dynamic friction angle is specified. A factor =1 is equivalent "
+                           "to equation 13 in van Dinther et al., (2013, JGR). A factor between 0 and 1 makes the "
+                           "curve of the friction angle vs. the strain rate more smooth, while a factor <1 makes "
+                           "the change between static and dynamic friction angle more steplike. "
+                           "Units: none.");
+
+        // rate and state parameters
+        prm.declare_entry ("Rate and state parameters a", "None",
```

Comment:
I see what you mean. Actually this can only be "Function" once I delete the enum, which I do not need.  So how do I set the default function to just be 1. or 0. everywhere or something similar?

Reply:
I think this parameter can go then. In the `Functions::ParsedFunction<dim>::declare_parameters(prm,1)`, the function expression will be set to zero by default.

Reply:
BTW, you could also use one function with two components, one for a and one for b. Not sure if it simplifies things much.

Reply:
cool. right now I am using this parameter also in:
       if ( prm.get("Rate and state parameter a") == "Function" )
          {
            prm.enter_subsection("Rate and state parameter a function");

what should I do here instead then? Just leave the if out?